sdk/core/Azure.Core/tests/HttpPipelineRequestContentTests.cs

```diff
@@ -110,14 +110,14 @@ public void DynamicDataContent()
                 """u8;
             ReadOnlyMemory<byte> json = new ReadOnlyMemory<byte>(utf8Json.ToArray());
 
-            JsonDocument doc = JsonDocument.Parse(json);
+            using JsonDocument doc = JsonDocument.Parse(json);
             MemoryStream expected = new();
```

Comment:
Nit: this should have a `using` too. Any `IDisposable` that we don't return (don't own the lifetime) should. I'm curious, are you not seeing compiler errors or warnings on this? Wonder if we should make sure it is erring because it can lead to spikes of GC especially problematic in server scenarios.

Again, though, this is in tests but is a good practice. The comment about errors was more for any production code we ship.

Reply:
Same below for the `MemoryStream`.

Reply:
There's a few CA rules like CA2213, but CA2000 is what appears to be most relevant: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2000

This is disabled by default (severity: off). I could swear I've seen it on before. If we want to enable it (and others), we should consider doing so via .editorconfig to keep everything in a single file, though we also have _eng/CodeAnalysis.ruleset_ which is *currently* used for StyleCop, but doesn't have to be limited to that. In fact, I wonder if we should get rid of that in lieu of .editorconfig that is used by more things: VS, VSCode (with extension, though maybe OmniSharp uses it by default), vim (with extension), etc.

I'll open a separate issue to consider all this.